jvm-agent/build.gradle.kts

```diff
@@ -31,9 +35,26 @@ sourceSets {
         api("org.ow2.asm:asm-util:${asmVersion}")
         api("net.bytebuddy:byte-buddy:${byteBuddyVersion}")
         api("net.bytebuddy:byte-buddy-agent:${byteBuddyVersion}")
+
+        val junitVersion: String by project
+
+        testImplementation("junit:junit:$junitVersion")
     }
 }
 
+tasks.withType<Test> {
+        javaLauncher.set(
+            javaToolchains.launcherFor {
+                val jdkToolchainVersion: String by project
+                val testInTraceDebuggerMode: String by project
+                val jdkVersion = jdkToolchainVersion.toInt()
+                // https://github.com/JetBrains/lincheck/issues/500
+                val jreVersion = if (testInTraceDebuggerMode.toBoolean() && jdkVersion == 8) 17 else jdkVersion
+                languageVersion.set(JavaLanguageVersion.of(jreVersion))
+            }
+        )
+    }
```

Comment:
It looks like this was code was copied from top-level `build.gradle.kts`. 
Could you please instead extract into a method and put into `buildSrc/main/kotlin/Testing.kt`?